### PR TITLE
Submit CMS changes to API

### PR DESCRIPTION
**Pull request by laurensgroeneveld**
Sync saved local changes with API when online

Uses hardcoded groups (probably needs to be solved in another PR when we can get the groups from the API)

### CODE REVIEW

File: api/src/dto/_baseDto.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export class _baseDto {
     @IsEnum(DocType)
     type: DocType;
 
-    @IsDate()
+    // TODO check if it's a valid timestamp
```
Comment (laurensgroeneveld) [inline api/src/dto/_baseDto.ts:20]: When I posted it with a timestamp I got an error "needs to be instance of Date"

Comment (ivanslabbert) [reply api/src/dto/_baseDto.ts:20]: Might need to add @IsNumber to prevent sending strings etc?

File: api/src/dto/ContentDto.ts [hunk 2]
```diff
@@ -51,10 +53,11 @@ export class ContentDto extends _contentBaseDto {
     video?: Uuid;
 
     @IsDate()
-    @IsNotEmpty()
+    @IsOptional()
```
Comment (ivanslabbert) [inline api/src/dto/ContentDto.ts:56]: I think we can require it to be set before saving but by default set it to the document creation date (in the CMS)? The editor should be able to set it to a different date if he wants to.

Will we only include the document in Sync after the publish date? (perhaps not in MVP but later) to enable "scheduled publish".

File: api/src/dto/ContentDto.ts [hunk 3]
```diff
@@ -51,10 +53,11 @@ export class ContentDto extends _contentBaseDto {
...
     publishDate: number;
 
     @IsDate()
+    // TODO required if status set to published
```
Comment (ivanslabbert) [inline api/src/dto/ContentDto.ts:60]: Do not think we need to have the expiry date set, unless you want the document to expire?

Notes for CMS functionality:
Would be nice if it is possible to both be able to set a date or number of days.

File: api/src/dto/_baseDto.ts (hunk 1 above)
Comment (ivanslabbert) [inline api/src/dto/_baseDto.ts:20]: Set to @IsNumber for now (unless there is a better way to check if it is a Unix timestamp). Could perhaps check if the number is bigger than 1 January 2020 or so?

File: api/src/socketio.ts [hunk 4]
```diff
@@ -23,6 +23,7 @@ import { LanguageDto } from "./dto/LanguageDto";
 import { PostDto } from "./dto/PostDto";
 import { TagDto } from "./dto/TagDto";
 import { UserDto } from "./dto/UserDto";
+import { _baseDto } from "./dto/_baseDto";
```
Comment (ivanslabbert) [inline api/src/socketio.ts:26]: Not sure why we need to import _baseDto?

**Comment by ivanslabbert** [comment]
This PR might have some conflicts with PR #36

File: api/src/dto/ContentDto.ts (hunk 2 above)
Comment (laurensgroeneveld) [reply api/src/dto/ContentDto.ts:56]: My thinking was that the document might not always be published the same day it was created, and that we can instead provide a dropdown in the CMS with options:
- Set to current date when published
- Custom date.. -> datepicker

So similar to how we said we would handle expiry date.

Then it's also handled for translations created after the original translation was published, and we avoid the problem of the current site where it defaults to a wrong (older) date.

File: api/src/dto/ContentDto.ts (hunk 3 above)
Comment (laurensgroeneveld) [reply api/src/dto/ContentDto.ts:60]: > Do not think we need to have the expiry date set, unless you want the document to expire?

Right, this comment belongs to publishedDate instead.

**Pull request merged**